frontend/packages/ux-editor/src/components/config/componentSpecificContent/Summary2/Override/Summary2OverrideDisplaytype.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { StudioCard, StudioNativeSelect } from '@studio/components';
+import type {
+  Summary2OverrideConfig,
+  SummaryCustomTargetType,
+} from 'app-shared/types/ComponentSpecificConfig';
+import { useCustomConfigType } from './hook/useCustomConfigType';
+import { useTranslation } from 'react-i18next';
+
+export interface Summary2OverrideDisplaytypeProps {
+  override: Summary2OverrideConfig;
+  onChange: (override: Summary2OverrideConfig) => void;
+}
+
+export const Summary2OverrideDisplaytype = ({
+  override,
+  onChange,
+}: Summary2OverrideDisplaytypeProps) => {
+  const customConfigType = useCustomConfigType();
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    if (!override.displayType) {
+      onChange({ ...override, displayType: 'string' });
+    }
+  }, [override, onChange]);
```

Comment:
Maybe we can convert this to a hook to make the code more readable? 😄

Reply:
What about adding just this check and remove useEffect ? It worked fine. 
```suggestion
 if (!override.displayType) {
    const updatedOverrideType = { ...override, displayType: 'string' } as Summary2OverrideConfig;
    onChange(updatedOverrideType);
    return null;
  }
```

Reply:
What I was thinking about initially was to split the code up to increase readability and separation of concerns 😄 
But now that I have looked more at it, why do we need this if statement in the first place? 🤔

Reply:
To set the initial value to a string when the display type appears after selecting a checkbox or multiple select, we can remove it if not necessary ?

Reply:
@JamalAlabdullah, I may not fully grasp your code, but I’m questioning the necessity of the `useEffect`. From my perspective, it seems we could remove it. I've outlined my reasoning in the figure below, so please take a look at the points from 1 to 3.

![image](https://github.com/user-attachments/assets/c865e649-15f3-47e2-b318-d0fecb2f06a1)

Reply:
Yes, the useEffect was removed some time ago 🙂, but I agree with your suggestion—it makes sense 🙂